edit.js

```diff
@@ -12,10 +14,73 @@ appliesToEverythingTemplate.innerHTML = t("appliesToEverything") + ' <button cla
 var sectionTemplate = document.createElement("div");
 sectionTemplate.innerHTML = '<label>' + t('sectionCode') + '</label><textarea class="code"></textarea><br><div class="applies-to"><label>' + t("appliesLabel") + ' <img class="applies-to-help" src="help.png" alt="' + t('helpAlt') + '"></label><ul class="applies-to-list"></ul></div><button class="remove-section">' + t('sectionRemove') + '</button><button class="add-section">' + t('sectionAdd') + '</button>';
 
+document.addEventListener("change", function(event) {
+	var node = event.target;
+	if (node.type && !node.form) { // INPUTs that aren't in a FORM are stylesheet
+		switch (node.type) {
+			case "checkbox":
+				setCleanItem(node, node.checked === node.defaultChecked);
+				break;
+			case "text":
+			case "select-one":
+			case "select-multiple":
+				setCleanItem(node, node.value === node.defaultValue);
+				break;
+		}
+	}
+});
+
+// Set .dirty on stylesheet contributors that have changed
+var items = {};
+function isCleanItem(node) {
+	return items[node.id];
+}
+function setCleanItem(node, clean) {
+	var id = node.id;
+	if (!id) id = node.id = Date.now().toString(32).substr(-6);
+	items[id] = clean;
+
+	if (clean) node.classList.remove("dirty");
+	else node.classList.add("dirty");
```

Comment:
I prefer braces and multi-line `if`s, but I wouldn't reject a PR if it had code in another format.